### DISCUSSION TITLE
Suspected bug: SENDMAIL Crashing on certain SNMP server

### DISCUSSION THREAD

**Discussion by UBWH** [arendst/Tasmota#20640] [General]
See below some examples of using the sendmail command line command
(ESP32)

## Example 1 ##
74 char command
The values are indeed wrong (bogus user/password, etc)
```
06:28:50.528 CMD: sendmail [hostgator.com:465:Longuser:bigpassword:test@test.com.au:test@xyz.com:alarm] test
06:28:50.534 CMD: Grp 0, Cmd 'SENDMAIL', Idx 1, Len 81, Pld -99, Data '[hostgator.com:465:Longuser:bigpassword:test@test.com.au:test@xyz.com:alarm] test'
06:28:50.537 MAI: Size 74
06:28:50.539 MAI: hostgator.com, 465, Longuser, bigpassword
06:28:50.542 MAI: test@test.com.au, test@xyz.com, alarm,  test
06:29:24.770 RSL: RESULT = {"Sendmail":"Wrong parameters"}
```

## Example 2 ##
71 char command

Tasmota crashes
```
sendmail [gator3233.hostgator.com:465:user:pw:test@test.com.au:test@xyz.com:alarm] test

06:30:54.175 MQT: tele/solar_06C6B8/INFO1 = {"Info1":{"Module":"Solar","Version":"13.2.0(tasmota32)","FallbackTopic":"cmnd/DVES_06C6B8_fb/","GroupTopic":"cmnd/tasmotas/"}}
06:30:54.200 MQT: tele/solar_06C6B8/INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"solar-06C6B8-1720","IPAddress":"10.1.1.8","IP6Global":"fdfd:f5d8:e1fd::b2a7:32ff:fe06:c6b8","IP6Local":"fe80::b2a7:32ff:fe06:c6b8"}}
06:30:54.224 MQT: tele/solar_06C6B8/INFO3 = {"Info3":{"RestartReason":{"Exception":1,"Reason":"Syscall","EPC":"400876fc","EXCVADDR":"00000000","CallChain":["400876f9","3ffb03cd"]},"BootCount":320}}
```

## Comments ##
I have tried all sorts of combinations for my sendmail command, and is seems the snmp server of `gator3233.hostgator.com` is the problem.

This really is my SNMP server 
![image](https://github.com/arendst/Tasmota/assets/72185209/2a1dcbbd-ca3b-4c49-b743-24401b5914ba)

Naturally I tried `sendmail [xxx] test` with my real username, password, sender email, receiver email, etc. It crashes also.

Any suggestions?

**Comment by UBWH** [comment]
BTW - HEAP was over 100MB when sendmail command issued

**Reply by gemu2015** [reply]
did you check this ?

https://github.com/arendst/Tasmota/discussions/20588

**Reply by UBWH** [reply]
:-)

Success. This changed fixed the crash. Thanks.
```
#undef SET_ESP32_STACK_SIZE
#define SET_ESP32_STACK_SIZE (12 * 1024)
```
